### DISCUSSION TITLE
Horizontal scrolling in tables

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#2950] [General] [answered]
[Redacted]

Non-public information

**Comment by loreina** [comment]
Hi @Fazle-Jahangir, I brought this back to the team and we'll reply with a follow up shortly. Thanks for your patience!

**Reply by Fazle-Jahangir** [reply]
[Redacted]

Non-public information

**Comment by loreina** [comment, marked as answer]
Hi @Fazle-Jahangir,

Thanks for the prototype, it was really helpful to understand the intended interaction.

This looks like the use case for a Tree Grid since there's collapsible rows. We don’t currently have that in Paste but it’s on our component backlog. We don’t have an ETA for when it will be available unfortunately. You may need to roll your own solution or adopt a third party library for this particular component.

You could also consider using [Data Grid](https://paste.twilio.design/components/data-grid). It has horizontal scrolling and sticky headers, though it doesn't currently have frozen columns. You can submit a [dependency request](https://airtable.com/shrvMvL3BG6Zm0f9l) for adding frozen columns to Data Grid and the team will plan to prioritize it in our roadmap.

Let us know what you end up doing, would be curious to hear what you landed on.
